### DISCUSSION TITLE
Where should be put our fozzie test files?

### DISCUSSION THREAD

**Discussion by oliversweb** [justeattakeaway/fozzie-components#1171] [General]
What should be the standard?. do we put all the fozzie unit tests in a tests folder off of root or do we create a tests folder next to each file/s we are testing and add the unit tests to it?

_Take into consideration: we most probably have a test helper file/s that are imported into most test files for test data, etc. plus the length of import paths_

**Comment by DamianMullins** [comment]
[We've had this discussion in the past](https://github.com/orgs/justeat/teams/ui-developers-guild/discussions/1)  and the consensus then was to place unit tests close to the unit under test in a `_tests` directory.

**Reply by DamianMullins** [reply]
This would still be my preference otherwise you end up either replicating the folder structure for the tests or get into a muddle if all files are in a single directory and multiple units have the same filename (think `index.js` etc). 

Having the tests close to the unit under test also makes them really easy to find.

**Comment by simonsandersje** [comment]
Agree with @DamianMullins - unit tests should reside close to the logic under test, as you are often editing the logic/unit tests together.

**Comment by oliversweb** [comment]
Where would you put the test helper files that are used by most/all the tests?

**Reply by DamianMullins** [reply]
Also [mentioned in that older thread](https://github.com/orgs/justeat/teams/ui-developers-guild/discussions/1/comments/16) — global util/helpers should go at the root of the project (or the root of the front-end code directory).
